components/output-component-factory.js

```diff
@@ -0,0 +1,74 @@
+// output-component-factory.js
+
+var _ = require('underscore');
+var componentFactory = require('./change-component-factory');
+
+/**
+ * Captures thrown errors from the onchange handler and sends them to the
+ * 'error' port, and sends the result (maybe a Promise) of the handler function to
+ * the 'output' port.
+ *
+ * Usage:
+ *  componentFactory({
+ *      inPorts: {input:{indexBy:'vnid'}} || ['input'],
+ *      onchange: function({portName:payload})
+ *  });
+ */
+module.exports = function(nodeDef){
+    return componentFactory(_.defaults({
+        inPorts: _.isArray(nodeDef.inPorts) ?
+            _.object(nodeDef.inPorts, _.times(nodeDef.inPorts.length, _.object)) :
+            nodeDef.inPorts,
+        outPorts: _.extend({
```

Comment:
Should this be _.default?   The updater developer could choose to define their own.